lxd/ubuntu-pro/watcher.go

```diff
@@ -0,0 +1,252 @@
+package pro
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/logger"
+)
+
+// GuestAttachSetting represents ubuntu-pro settings for guest attachment availability.
+type GuestAttachSetting string
+
+const (
+	// GuestAttachSettingOff indicates that guest attachment is never available. The value of `GET /1.0/ubuntu-pro` over
+	// devlxd should always be "off".
+	GuestAttachSettingOff GuestAttachSetting = "off"
+
+	// GuestAttachSettingAvailable indicates that guest attachment is available and can be opted into via guest config.
+	// The value of `GET /1.0/ubuntu-pro` over devlxd should return the value of the `ubuntu_pro.guest_attach`
+	// configuration of the instance.
```

Comment:
We should mention here that its not only the instance configuration setting that influences the return value of `GET /1.0/ubuntu-pro` and that its also the status of the pro client on the LXD host too.